base/array.jl

```diff
@@ -26,11 +26,13 @@ typealias DenseVecOrMat{T} Union{DenseVector{T}, DenseMatrix{T}}
 import Core: arraysize, arrayset, arrayref
 
 """
-    Array{T,N}(dims)
+    Array{T}(dims)
```

Comment:
It might be nice to have both signatures at the top, i.e.
```julia
    Array{T}(dims)
    Array{T,N}(dims)

Construct...
```
I should have done that in my previous PR.